### ISSUE TITLE
dimensioning & create vm's: voctos, streambackend, streamfrontend

### ISSUE THREAD

**Issue by markvdb** [FOSDEM/video#152]
- DC's:
  -  in order of preference: Nuremberg, Falkenstein, Helsinki
  - all in the same DC

- 60 voctos: hetzner ccx21 (4 reserved vcpu)
- 1 streambackend:
  - AX161: AMD Epyc 7502p 32 core "Rome", 128 Gb RAM, 2x 3.84 Tb NVMe SSD, 10 Gbit uplink option
  - (2020):
    -  hetzner ccx11 (2 reserved vcpu)
    -  video-backend-data: hetzner block storage 1 Tb
- 10 streamfrontends: hetzner cx21 (2 non-reserved vcpu) + **room to scale this up  if necessary**
- 1 streamdump0: ccx11 (2 reserved vcpu)+ hetzner block storage
- 0 streamdump-external0:  not bothering for this edition
- 1 web0: hetzner cx21
- 6 encoders: hetzner cpx41 (last year: cx51)
- 1 encode-master: hetzner cx21
- 1 reviewstorage: hetzner cx21
- 1 review-volume: hetzner block storage 2 Tb (2020: 1 Tb)

TOTALS:
- 1 ccx11
- 60 ccx21
-  1 AX161+ 2x3.84 Tb NVMe SSD, 10 Gbit uplink
- 13 (to 24) cx21
- 6 cpx41

TOTALS dedicated/reserved/non-reserved instances
- 1 dedicated
- 61 reserved cpu instances
- 30 non-reserved cpu instances

**Comment by yoe** [comment]
DebConf used CPX41, FWIW

**Comment by markvdb** [comment]
That will certainly work... almost all of the time. I'd rather not take the risk of overloaded voctos though. That would be penny wise, pound foolish...

**Comment by yoe** [comment]
hence the "FWIW". DebConf didn't have any trouble using the CPX41 (it was using less than 50% CPU for most of the conference), but I definitely agree with "reserved CPUs is safer".

**Comment by markvdb** [comment]
@krokodilerian @gerryd I ran a quick comparison to 2020. Comments welcome!

Regarding the number of mixers, I'm being a bit generous here to simplify. Some rooms span two days with the same name. Most are single day. This simplifies things quite a bit, at the expense of a few hundered € worst case. Well worth it if you ask me...

**Comment by markvdb** [comment]
@gerryd Dimensioning updates now complete. Talked through with @krokodilerian and @yoe . Can you please ask Hetzner for the limit upgrade asap so I can start setting these up?

**Comment by gerryd** [comment]
> @gerryd Dimensioning updates now complete. Talked through with @krokodilerian and @yoe . Can you please ask Hetzner for the limit upgrade asap so I can start setting these up?

I have requested the limit upgrade, hopefully we will hear back from them soon.

I noticed you are intending to use a dedicated server as the stream backend machine. That should work, as it seems they have added the possibility to link "cloud networks" used for their cloud instances to the "vswitch networks" for dedicated machines (this wasn't the case yet last year).

However, their dedicated machines are only available in Falkenstein and Helsinki, as far as I can see. You will probably want to reconsider the Nuremberg option as a primary location.

**Comment by gerryd** [comment]
> > @gerryd Dimensioning updates now complete. Talked through with @krokodilerian and @yoe . Can you please ask Hetzner for the limit upgrade asap so I can start setting these up?
> 
> I have requested the limit upgrade, hopefully we will hear back from them soon.

The limits have been adjusted as requested.

**Comment by markvdb** [comment]
@corosuske Can you please create the majority of the vm's already, and the physical machine? Some hints:
- Make sure they're all in the same DC (Falkenstein).
- Name them in hetzner with their FOSDEM dns hostnames.
  - voctos https://github.com/FOSDEM/infrastructure/blob/master/ansible/hosts#L157-L208
  - streamfrontends, streambackend, streamdump0 https://github.com/FOSDEM/infrastructure/blob/master/ansible/hosts#L38-L53 , adding 5 streamfrontends to get 10  this year.
- Ask server@ for dns entries. server@ has asked for bind compatible A records to make this easier.
- Please make sure that they're all connected to the same existing private network.

@yoe Could you set up all sreview related machines? As discussed, the list at the top here is perfectly the same as you had last year, with the exception of somewhat cheaper but more powerful AMD based encoders.

Urgent questions or remarks? Please get in touch!

**Comment by corosuske** [comment]
just added the vocto machines for the devrooms can someone verify that these are created correctly 


dcomposition-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
188.34.163.108

dcicd-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
188.34.163.107

dcollab-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
188.34.163.109

dcad-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
188.34.163.106

dcloud-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
188.34.157.23

dcontainers-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
188.34.160.72

dblockchain-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
188.34.158.247

dbsd-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
188.34.160.121

dcommunity-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
188.34.162.20

dapacheopenoffice-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
188.34.163.110

ddependency-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
88.198.77.237

dgeospatial-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
88.99.13.210

demulator-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
159.69.124.34

dgo-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
168.119.55.196

ddeclarativeminimalistic-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
168.119.53.106

ddesign-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
159.69.242.16

ddocs-vocto.video.fosdem.orgfsn1-5
CCX21 / 160 GB / fsn1-dc14
116.202.105.179

dembedded-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
78.47.232.217

dfirmware-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
116.202.22.209

ddistributions-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
88.198.153.110

dlegal-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
188.34.163.114

dlibreoffice-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
168.119.156.34

dmicrokernel-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
188.34.162.15

dkotlin-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
168.119.251.124

dhpc-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
188.34.163.113

djavascript-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
159.69.123.170

dmariadb-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
188.34.163.111

dinfra-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
188.34.163.112

dhardwaretrusted-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
188.34.162.43

dgraph-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
168.119.182.219

dmysql-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
188.34.163.125

dpostgresql-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
188.34.163.126

monitoring-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
188.34.163.118

dmozilla-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
188.34.163.119

dpython-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
188.34.163.122

dopenjdk-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
188.34.163.117

dpower-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
188.34.163.123

dperl-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
188.34.163.120

dnetwork-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
188.34.163.124

dopenchain-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
188.34.163.121

dretro-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
188.34.163.128

dwebperformance-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
138.201.156.22

dsds-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
188.34.162.19

dsafety-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
188.34.163.130

dvirtualization-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
159.69.49.135

dsdn-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
188.34.163.132

dradio-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
188.34.157.33

drtc-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
188.34.163.131

dtesting-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
188.34.160.5

dresearch-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
188.34.163.129

llightningtalks-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
188.34.163.136

dzig-vocto.video.fosdem.org
CCX21 / 160 GB / fsn1-dc14
188.34.163.137

**Comment by markvdb** [comment]
I can get into the first and last in the list. Thx Wouter!

**Comment by corosuske** [comment]
streamfrontend9.video.fosdem.org
CX21 / 40 GB / fsn1-dc14
168.119.103.127

streamfrontend3.video.fosdem.org
CX21 / 40 GB / fsn1-dc14
168.119.57.144

streamfrontend6.video.fosdem.org
CX21 / 40 GB / fsn1-dc14
168.119.254.28

streamfrontend8.video.fosdem.org
CX21 / 40 GB / fsn1-dc14
168.119.252.156

streamfrontend0.video.fosdem.org
CX21 / 40 GB / fsn1-dc14
168.119.255.221

streamfrontend7.video.fosdem.org
CX21 / 40 GB / fsn1-dc14
168.119.101.168

streamfrontend4.video.fosdem.org
CX21 / 40 GB / fsn1-dc14
168.119.109.31

streamfrontend2.video.fosdem.org
CX21 / 40 GB / fsn1-dc14
168.119.176.163

streamfrontend1.video.fosdem.org
CX21 / 40 GB / fsn1-dc14
188.34.157.110

streamfrontend5.video.fosdem.org
CX21 / 40 GB / fsn1-dc14
188.34.176.113

**Comment by corosuske** [comment]
streamfrontend0.video.fosdem.org
CX21 / 40 GB / fsn1-dc14
168.119.255.221

streamfrontend1.video.fosdem.org
CX21 / 40 GB / fsn1-dc14
188.34.157.110

streamfrontend2.video.fosdem.org
CX21 / 40 GB / fsn1-dc14
168.119.176.163

streamfrontend3.video.fosdem.org
CX21 / 40 GB / fsn1-dc14
168.119.57.144

streamfrontend4.video.fosdem.org
CX21 / 40 GB / fsn1-dc14
168.119.109.31

streamfrontend5.video.fosdem.org
CX21 / 40 GB / fsn1-dc14
188.34.176.113

streamfrontend6.video.fosdem.org
CX21 / 40 GB / fsn1-dc14
168.119.254.28

streamfrontend7.video.fosdem.org
CX21 / 40 GB / fsn1-dc14
168.119.101.168

streamfrontend8.video.fosdem.org
CX21 / 40 GB / fsn1-dc14
168.119.252.156

streamfrontend9.video.fosdem.org
CX21 / 40 GB / fsn1-dc14
168.119.103.127

**Comment by corosuske** [comment]
streamdump0.video.fosdem.org
CCX11 / 80 GB / fsn1-dc14
188.34.154.197

**Comment by corosuske** [comment]
web0.video.fosdem.org
CX21 / 40 GB / fsn1-dc14
188.34.155.122

**Comment by corosuske** [comment]
cfg encoder machines and storage for review ,  @yoe  do you want me to roll those out ?

**Comment by yoe** [comment]
I'll do that myself, thanks. Ansible configures most of the system, but not everything, and it requires some handholding to get things up and running.

**Comment by gerryd** [comment]
>     * 1 streambackend:
>       * AX161: AMD Epyc 7502p 32 core "Rome", 128 Gb RAM, 2x 3.84 Tb NVMe SSD, 10 Gbit uplink option

This one has been set up as well (took a bit longer as it is a custom dedicated server), the 10 Gbit NIC has also been installed.

It is available at `streambackend0.video.fosdem.org` with a minimal Debian buster install. I've also deployed the common role so all relevant users should have access.

I've set up a vswitch and attached it to the virtual machine network. Do note that the dedicated machine lives in a different subnet (10.21.1.0/24) from the VM network (10.21.0.0/24), and traffic is being routed. The dedicated machine has internal IP 10.21.1.2.

Finally, I have updated all DNS entries to be in sync with the current status at Hetzner (I noticed all IPv6 addresses for the virtual machines were wrong, for one). The `stream.fosdem.org` hostname now does round-robin between all frontends. And reverse DNS has been set up as well, for the finishing touch. :)

Pro tip: the Hetzner API works really well, the "official" client is also pretty decent. See [here](https://community.hetzner.com/tutorials/howto-hcloud-cli) and [here](https://github.com/hetznercloud/cli).
